### PR TITLE
Update to typescript, and publish the types

### PR DESCRIPTION
**Pull request by DavidTanner**
why not

It will be nice to have this project available with types.

### CODE REVIEW

**Comment by DavidTanner** [comment]
Sorry for the noise.  Tests passing now.

**Comment by xdumaine** [comment]
omg

File: package.json [hunk 1]
```diff
@@ -1,13 +1,15 @@
 {
   "name": "@lifeomic/axios-fetch",
-  "version": "1.5.2",
+  "version": "1.6.0",
```
Comment (xdumaine) [inline package.json:3]: I'm always hesitant to make major refactors a minor bump, so I'd vote for 2.x but not strongly enough to block on it.

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+export interface FetchInit extends Record<string, any> {
+  headers?: Record<string, string>;
+  method?: AxiosRequestConfig['method'];
+  body?: FormData | any;
+  extra?: any;
+}
+
+type AxiosTransformer = (config: AxiosRequestConfig, input: string | undefined, init: FetchInit) => AxiosRequestConfig;
+
+type AxiosFetch = (input?: string, init?: FetchInit) => Response;
```
Comment (xdumaine) [inline src/index.ts:16]: Will a consumer want to import this type? Is it worth exporting just in case?

**Review by xdumaine — DISMISSED** [review]

File: package.json [hunk 3]
```diff
@@ -16,24 +18,41 @@
...
     "nock": "^11.3.3",
     "nyc": "^15.0.0",
-    "sinon": "^7.2.4"
+    "sinon": "^10.0.1",
+    "ts-node": "^9.1.1",
+    "typescript": "^4.2.4"
   },
   "dependencies": {
-    "form-data": "^2.3.3",
+    "@types/lodash.identity": "^3.0.6",
```
Comment (mdlavin) [inline package.json:49]: Can we move these into the `devDependencies`? I have seen @types in the dependency list when the types are re-exported, but it doesn't seem like all of these types will be exported

**Review by mdlavin — DISMISSED** [review]

**Review by IevgenRagulin — APPROVED** [review]

**Review by mdlavin — APPROVED** [review]

File: src/index.ts (hunk 2 above)
Comment (DavidTanner) [reply src/index.ts]: Exported

File: package.json (hunk 3 above)
Comment (DavidTanner) [reply package.json]: I've had issues where the type was used internally and not exported, so no I export any types used in lib code.

File: package.json (hunk 1 above)
Comment (DavidTanner) [reply package.json]: Bumped to 2.0.0

**Pull request merged**